### PR TITLE
Fixes #4215 "Swiping on the last item in explore featured tab takes you to uploaded via mobile"

### LINKED ISSUE #4215: Swiping on the last item in explore featured tab takes you to uploaded via mobile.
**Summary:** 

When you swipe on the last item in explore featured fragment it gets to uploaded via mobile.

**Steps to reproduce:** 
go to the last item of the featured tab and swipe left, you'll notice you're taken to uploaded via mobile.
* this issue also exist in bookmark fragment.

What did you expect the app to do, and what did you see instead?

This is not the major issue but when we click on media in this uploaded via mobile tab, the app shows some unexpected behavior when returning from the media details.

**Device and Android version:** 

Nokia 6.1 android version 10 (stock).
**Commons app version:** 

Latest master.

**Screen recording**

ezgif com-resize

Observations:
* First we can see that 'Uploaded from mobile/featured' tabs are not there.
* Media details work perfectly fine.
* On returning from details 'Uploaded from mobile/featured' tabs are restored.
* On switching to featured we see that the list fragment is replaced by the view pager which does not allow vertical scroll between items although scrolling horizontally is possible as in view pager.

**Would you like to work on the issue?**

Please let us know whether you want to fix the issue by yourself. If not, anyone can get the issue assigned to them.
Yes.

### PR DESCRIPTION
**Pull request by 4D17Y4**
**Description (required)**

Fixes #4215

What changes did you make and why?
* Had to add an extra class ParentViewPager which is a custom View pager whose scrolling can be enabled and disabled.
* changed the view pagers to parent view pager.
* Disabled the swipe on entering the details fragment and enabling it on exiting.

**Tests performed (required)**

Tested betaDebug on Nokia 6.1 + with API level 29.

### CODE REVIEW

**Comment by 4D17Y4** [comment]
I was not sure where to keep the ParentViewPager Class, so temporarily kept it in explore.

This PR fixes the issue both in the explore and bookmarks fragment.

Up for review 
@misaochan , @neslihanturan , @madhurgupta10 .

**Comment by nicolas-raoul** [comment]
I just tested this PR:

On the "Featured" tab I opened the details of a picture, then I swiped left very fast for over 1 minute. Unlike on master, I always saw featured pictures (easily recognizable even when swiping very fast because they all have many categories) and did not start seeing pictures from "uploaded via mobile".

In other words, the PR seems to fix the issue :-)

File: app/src/main/java/fr/free/nrw/commons/explore/ParentViewPager.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package fr.free.nrw.commons.explore;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.MotionEvent;
+import androidx.viewpager.widget.ViewPager;
+
+public class ParentViewPager extends ViewPager {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/explore/ParentViewPager.java:8]: Please add some Javadoc to explain what is the purpose of this class, for instance you can explain the bug briefly and say that how this class helps prevent it.
Also, Javadoc for each member and method, even if a super simple one, is our project rule. Thanks for your understanding :-)

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Otherwise looks good to me

File: app/src/main/java/fr/free/nrw/commons/explore/ParentViewPager.java (hunk 1 above)
Comment (4D17Y4) [reply app/src/main/java/fr/free/nrw/commons/explore/ParentViewPager.java:10]: Javadoc added.

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
The javadoc content is great!
Last thing, the standard format for Javadoc comments is this one:
```
/**
* Example.
*/
 ```
It is not only stylistic: This format is recognized and used by automatic documentation generators, and by IDEs to show useful tooltips/completion info/etc.
https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html

**Comment by 4D17Y4** [comment]
Javadoc updated.
Thx @nicolas-raoul,
I will keep this in mind.

**Review by nicolas-raoul — APPROVED** [review]

**Pull request merged**